x-pack/platform/test/fleet_api_integration/apis/agents/list.ts

```diff
@@ -143,22 +143,57 @@ export default function ({ getService }: FtrProviderContext) {
     });
 
     it('should return metrics if available and called with withMetrics', async () => {
+      await es.index({
+        id: 'agent-without-metrics',
+        index: '.fleet-agents',
+        refresh: 'wait_for',
+        document: {
+          access_api_key_id: 'api-key-2',
+          active: true,
+          policy_id: 'policy1',
+          type: 'PERMANENT',
+          local_metadata: { host: { hostname: 'host2' } },
+          user_provided_metadata: {},
+          enrolled_at: '2022-06-21T12:14:25Z',
+          last_checkin: '2022-06-27T12:27:29Z',
+          tags: ['existingTag'],
+          agent: { id: 'agent-without-metrics', version: '9.2.0' },
+        },
+      });
+      await es.index({
+        id: 'agent-with-metrics',
+        index: '.fleet-agents',
+        refresh: 'wait_for',
+        document: {
+          access_api_key_id: 'api-key-1',
+          active: true,
+          policy_id: 'policy1',
+          type: 'PERMANENT',
+          local_metadata: { host: { hostname: 'host1' } },
+          user_provided_metadata: {},
+          enrolled_at: '2022-06-21T12:14:25Z',
+          last_checkin: '2022-06-27T12:27:29Z',
+          tags: ['existingTag'],
+          agent: { id: 'agent-with-metrics', version: '9.2.0' },
+        },
+      });
       const now = Date.now();
       // We need to create data points in precise time buckets to ensure the derivative works properly
-      // 4 minutes ago (first data point for component1)
-      const fourMinutesAgo = new Date(now - 4 * 60 * 1000);
-      fourMinutesAgo.setSeconds(0, 0); // Set to exact minute boundary
+      // 3 minutes ago (first data point for component1)
+      const threeMinutesAgo = new Date(now - 3 * 60 * 1000);
+      threeMinutesAgo.setSeconds(0, 0); // Set to exact minute boundary
+
       await es.index({
         index: 'metrics-elastic_agent.elastic_agent-default',
         refresh: 'wait_for',
         document: {
-          '@timestamp': fourMinutesAgo.toISOString(),
+          '@timestamp': threeMinutesAgo.toISOString(),
```

Comment:
While debugging I found that the first doc with timestamp four minutes ago didn't show up in the metrics search that checks last 5 minutes. Reducing the time range to three minutes ago seemed to help, it could be that there is some slowness during tests that caused the first doc to go out of range.